crates/pera-core/src/dwallet_mpc/dkg.rs

```diff
@@ -48,27 +54,57 @@ impl FirstDKGBytesParty {
     }
 }
 
+/// Deserializes the messages received from other parties for the next advancement.
+/// Any value that fails to deserialize is considered to be sent by a malicious party.
+/// Returns the deserialized messages or an error including the IDs of the malicious parties.
+pub fn deserialize_mpc_messages<M: DeserializeOwned + Clone>(
+    messages: HashMap<PartyID, Vec<u8>>,
+) -> PeraResult<HashMap<PartyID, M>> {
+    let parsing_results: Vec<PeraResult<(PartyID, _)>> = messages
+        .iter()
+        .map(|(k, v)| {
+            let value =
+                bcs::from_bytes(&v).map_err(|_| PeraError::DWalletMPCMaliciousParties(vec![*k]))?;
```

Comment:
Also, what is the reason for putting `*k` into a `vec![]` here?

Reply:
The DWalletMPCMaliciousParties error type receives a vec of malicious parties so it will be able to handle the case we have more than 1 malicious party